frontend/src/components/panels/Layers.svelte

```diff
@@ -424,10 +428,10 @@
 						<IconButton
 							class={"status-toggle"}
 							classes={{ inactive: !listing.entry.parentsUnlocked }}
-							action={(e) => (toggleLayerVisibility(listing.entry.id), e?.stopPropagation())}
+							action={(e) => (toggleLayerLock(listing.entry.id), e?.stopPropagation())}
 							size={24}
-							icon={listing.entry.parentsUnlocked ? "PadlockLocked" : "PadlockUnlocked"}
-							hoverIcon={listing.entry.parentsUnlocked ? "PadlockUnlocked" : "PadlockLocked"}
+							icon={(listing.entry.parentId && listing.entry.parentsUnlocked) || !listing.entry.unlocked ? "PadlockLocked" : "PadlockUnlocked"}
+							hoverIcon={(listing.entry.parentId && listing.entry.parentsUnlocked) || !listing.entry.unlocked ? "PadlockUnlocked" : "PadlockLocked"}
```

Comment:
I don't think we actually need the first part of the `||` clause? Just the change to `!listing.entry.unlocked` (my mistake for copy/pasting it to use `listing.entry.parentsUnlocked` when I first wrote this, thanks for the fix). Or does that first part of the `||` clause do something I'm not realizing?

Reply:
yeah, I think so, `listing.entry.parentsUnlocked` is just used for the classes, for marker inactive classes on that icon button.

Reply:
Thanks for the change. Sorry to nitpick but can you also remove the `!` and reverse the ternary clauses?

Reply:
Sure, I think that make easier to understand the operation

Reply:
Thanks.